src/pretty/mod.rs

```diff
@@ -427,7 +431,11 @@ impl PrettyPrinter {
             BoxDoc::nil(),
             (BoxDoc::text("{"), BoxDoc::text("}")),
             true,
-            FoldStyle::Never,
+            if codes.len() == 1 {
```

Comment:
The core change in this PR: if the code block contains only one item, we will put it in a single line if there are no new lines in that item. Am I correct?

Reply:
Yes. It just adds a special case for blocks with only one item.